src/libopensc/card-starcos.c

```diff
@@ -339,18 +341,6 @@ static int starcos_init(sc_card_t *card)
 		card->max_recv_size = 128;
 	}
 
-	if (sc_parse_ef_atr(card) == SC_SUCCESS) {
-		if (card->ef_atr->card_capabilities & ISO7816_CAP_EXTENDED_LENGTH) {
-			card->caps |= SC_CARD_CAP_APDU_EXT;
-		}
-		if (card->ef_atr->max_response_apdu > 0) {
-			card->max_recv_size = card->ef_atr->max_response_apdu;
-		}
-		if (card->ef_atr->max_command_apdu > 0) {
-			card->max_send_size = card->ef_atr->max_command_apdu;
-		}
-	}
```

Comment:
I had issues with extended APDUs in starcos_decipher(): some cards have reported an invalid length in EF.ATR and causing an APDU formatting error.  I review this problem again and put back the parsing of EF.ATR if possible.

Reply:
I have reapplied the function sc_parse_ef_atr(). I had to adjust the reported maximum size values to prevent error response 67 00 (Wrong length) in iso7816_read_binary() or during decryption.
Some card profiles reported the length limits only in the IAS/ECC specific issuer tag, I added parsing of this value as well.